### PR TITLE
Functional efficient path by refering to dB

### PR DESCRIPTION
**Pull request by alphakennyn**

### CODE REVIEW

**Comment by Theebz95** [comment]
The confirming action is only used when the user tries to select a destination? Is it used for anything else?

**Comment by alphakennyn** [comment]
@Theebz95 It's important to note that there's 2 types of selecting a destination: inspect and confirm.

-Inspect is trigger when the user taps once on a bus stop
-Confirm is trigger when the user double-tap on a bus stop (Can be changed if need be)

Only when the user confirms will a path be generated.

To answer your second question. No, it's not used anywhere else.

**Comment by michaelgoldfish** [comment]
The map.ts changes are just the version before the autocomplete worked so that's good.
So this new "formulas.ts" file gets used in google-maps.ts correct? If so that hopefully means implementing the search bar to send you somewhere wouldn't conflict with this new commit at all

File: mTransit/src/providers/google-maps.ts [hunk 1]
```diff
@@ -297,14 +297,13 @@ export class GoogleMaps {
 
     marker.addListener('click', () => {
     marker.setIcon(selectedMarker);
-    infowindow.open(this.map, marker);
+    //infowindow.open(this.map, marker);
```
Comment (ericpayette) [inline mTransit/src/providers/google-maps.ts:255]: There is some commented out code that should be deleted.

File: mTransit/src/providers/google-maps.ts [hunk 2]
```diff
@@ -471,6 +488,58 @@ export class GoogleMaps {
...
     }
+    for(let marker of this.busPath){
+      marker.setMap(null);
+    }
+    
     this.markers.length = 0;   
+    this.busPath.length = 0;       
+  }
+
+  loadShapes(lat: Number, lng: Number){
```
Comment (LordRoost) [inline mTransit/src/providers/google-maps.ts]: For this method, there is commented code on lines 510-512 and the comment on line 502 seems wrong.

**Review by zephix109 — COMMENTED** [review]
What does the formulas.ts file do?  Is the ApplyHaversine method the same as the one we used to have in the google-maps service?

File: mTransit/src/providers/google-maps.ts (hunk 1 above)
Comment (alphakennyn) [reply mTransit/src/providers/google-maps.ts:255]: In this example, I commented out infoWindow incase we ever use it again to "confirm" our users

**Comment by alphakennyn** [comment]
@zephix109  Since we want to find the nearest bus stop (that shares the same bus as our confirmed one), we have to find the nearest bus stop among that bus route. So I had thought to re-use the ApplyHaversine method just like we did to find all the bus stops near a clicked location. However the original bus stops and shapes have different name for their latitude and longitude attribute. So I just made another ApplyHaversine specifically for shapes.

Also, I had plan to put every long calculations in the formulas class, but that's to be decided later.

**Comment by zephix109** [comment]
Fair enough, I'll be refactoring all of the added code immediately.

**Review by zephix109 — APPROVED** [review]

**Pull request merged**